### PR TITLE
Feature al 796 add additional logic for limit orders inside mea

### PR DESCRIPTION
**Pull request by mjoksa**
https://lykkex.atlassian.net/browse/AL-796

### CODE REVIEW

File: client/Lykke.AlgoStore.MatchingEngineAdapter.Client/Lykke.AlgoStore.MatchingEngineAdapter.Client.csproj [hunk 1]
```diff
@@ -33,9 +33,12 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Lykke.AlgoStore.MatchingEngineAdapter.Abstractions" Version="1.0.22" />
     <PackageReference Include="Lykke.Common" Version="7.0.3" />
-    <PackageReference Include="protobuf-net" Version="2.3.7" />
+    <PackageReference Include="protobuf-net" Version="2.3.13" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Lykke.AlgoStore.MatchingEngineAdapter.Abstractions\Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.csproj" />
```
Comment (dafinkagerdjikova) [inline client/Lykke.AlgoStore.MatchingEngineAdapter.Client/Lykke.AlgoStore.MatchingEngineAdapter.Client.csproj]: We shouldn't use ProjectReference we should call the NuGet package

File: tests/Lykke.AlgoStore.MatchingEngineAdapter.Tests/Lykke.AlgoStore.MatchingEngineAdapter.Tests.csproj [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
+    <PackageReference Include="protobuf-net" Version="2.3.13" />
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\src\Lykke.AlgoStore.MatchingEngineAdapter.Abstractions\Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.csproj" />
     <ProjectReference Include="..\..\src\Lykke.AlgoStore.MatchingEngineAdapter.Services\Lykke.AlgoStore.MatchingEngineAdapter.Services.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="Lykke.AlgoStore.CSharp.AlgoTemplate.Models">
```
Comment (dafinkagerdjikova) [inline tests/Lykke.AlgoStore.MatchingEngineAdapter.Tests/Lykke.AlgoStore.MatchingEngineAdapter.Tests.csproj]: Again we need the NuGet package for Models project

**Review by dafinkagerdjikova — CHANGES_REQUESTED** [review]

File: src/Lykke.AlgoStore.MatchingEngineAdapter/Modules/MatchingEngineModule.cs [hunk 3]
```diff
@@ -1,34 +1,43 @@
...
         }
 
         protected override void Load(ContainerBuilder builder)
         {
-            builder.BindMeClient(_settings.CurrentValue.MatchingEngineClient.IpEndpoint.GetClientIpEndPoint().Result,
-                socketLog: null, ignoreErrors: true);
+            builder.RegisgterMeClient(_settings.CurrentValue.MatchingEngineClient.IpEndpoint.GetClientIpEndPoint());
+            //builder.RegisgterMeClient(_settings.CurrentValue.MatchingEngineClient.IpEndpoint.GetClientIpEndPoint());
+
+            //builder.Register(s =>
```
Comment (dafinkagerdjikova) [inline src/Lykke.AlgoStore.MatchingEngineAdapter/Modules/MatchingEngineModule.cs]: Remove this comments

**Review by dafinkagerdjikova — APPROVED** [review]

File: client/Lykke.AlgoStore.MatchingEngineAdapter.Client/MatchingEngineAdapterClient.cs [hunk 4]
```diff
@@ -58,5 +59,47 @@ public async Task<string> PingAsync(string content)
 
             return response;
         }
+
+        public async Task<ResponseModel<LimitOrderResponseModel>> PlaceLimitOrderAsync(string walletId, string assetPairId, OrderAction orderAction, double volume,
+            double price, string instanceId, bool cancelPreviousOrders = false)
+        {
+            var limitOrderRequest = new LimitOrderRequest
+            {
+                ClientId = walletId,
```
Comment (todorivanov) [inline client/Lykke.AlgoStore.MatchingEngineAdapter.Client/MatchingEngineAdapterClient.cs:68]: Should this field be named ClientId and then a walletId is assigned?

File: src/Lykke.AlgoStore.MatchingEngineAdapter.Services/MatchingEngineAdapter.cs [hunk 5]
```diff
@@ -20,51 +17,31 @@ public class MatchingEngineAdapter : IMatchingEngineAdapter
...
-            _feeCalculatorClient = feeCalculatorClient ?? throw new ArgumentNullException(nameof(feeCalculatorClient));
-            _assetsService = assetsService ?? throw new ArgumentNullException(nameof(assetsService));
-            _feeSettingsTargetClientIdHft = feeSettingsTargetClientIdHft ?? throw new ArgumentNullException(nameof(_feeSettingsTargetClientIdHft));
-            _log = log ?? throw new ArgumentNullException(nameof(log));
+            _feeCalculator = feeCalculator ?? throw new ArgumentNullException(nameof(feeCalculator));
+            _log = logFactory.CreateLog(this);
         }
 
-        public async Task<ResponseModel> CancelLimitOrderAsync(Guid limitOrderId)
+        public async Task<ResponseModel> CancelLimitOrderAsync(Guid limitOrderId, string instanceId)
```
Comment (todorivanov) [inline src/Lykke.AlgoStore.MatchingEngineAdapter.Services/MatchingEngineAdapter.cs]: Do you need this instanceId parameter? It's not used

File: src/Lykke.AlgoStore.MatchingEngineAdapter/Startup.cs [hunk 6]
```diff
@@ -56,30 +58,40 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
+                    settingsManager.ConnectionString(s => s.AlgoStoreMatchingEngineAdapter.Db.LogsConnectionString),
+                    "MatchingEngineAdapterLog",
+                    appSettings.CurrentValue.SlackNotifications.AzureQueue.ConnectionString,
+                    appSettings.CurrentValue.SlackNotifications.AzureQueue.QueueName);
+
                 var builder = new ContainerBuilder();
-                var appSettings = Configuration.LoadSettings<AppSettings>();
 
-                Log = CreateLogWithSlack(services, appSettings);
+                //Log = CreateLogWithSlack(services, appSettings);
```
Comment (todorivanov) [inline src/Lykke.AlgoStore.MatchingEngineAdapter/Startup.cs]: Remove these comments if they are not needed

**Review by todorivanov — APPROVED** [review]

**Pull request merged**